server/src/main/java/org/apache/gravitino/server/web/rest/OwnerOperations.java

```diff
@@ -0,0 +1,117 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.gravitino.server.web.rest;
+
+import com.codahale.metrics.annotation.ResponseMetered;
+import com.codahale.metrics.annotation.Timed;
+import java.util.Locale;
+import java.util.Optional;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import org.apache.gravitino.GravitinoEnv;
+import org.apache.gravitino.MetadataObject;
+import org.apache.gravitino.MetadataObjects;
+import org.apache.gravitino.authorization.Owner;
+import org.apache.gravitino.authorization.OwnerManager;
+import org.apache.gravitino.dto.requests.OwnerSetRequest;
+import org.apache.gravitino.dto.responses.MessageResponse;
+import org.apache.gravitino.dto.responses.OwnerResponse;
+import org.apache.gravitino.dto.responses.SetResponse;
+import org.apache.gravitino.dto.util.DTOConverters;
+import org.apache.gravitino.metrics.MetricNames;
+import org.apache.gravitino.server.authorization.NameBindings;
+import org.apache.gravitino.server.web.Utils;
+
+@NameBindings.AccessControlInterfaces
+@Path("/metalakes/{metalake}/owners")
+public class OwnerOperations {
+
+  private final OwnerManager ownershipManager;
+
+  @Context private HttpServletRequest httpRequest;
+
+  public OwnerOperations() {
+    // Because ownerManager may be null when Gravitino doesn't enable authorization,
+    // and Jersey injection doesn't support null value. So OwnerOperations chooses to retrieve
+    // ownerManager from GravitinoEnv instead of injection here.
+    this.ownershipManager = GravitinoEnv.getInstance().ownerManager();
+  }
+
+  @GET
+  @Path("{type}/{fullName}")
+  @Produces("application/vnd.gravitino.v1+json")
+  @Timed(name = "get-object-owner." + MetricNames.HTTP_PROCESS_DURATION, absolute = true)
+  @ResponseMetered(name = "get-object-owner", absolute = true)
+  public Response getOwnerForObject(
+      @PathParam("metalake") String metalake,
+      @PathParam("type") String type,
+      @PathParam("fullName") String fullName) {
+    try {
+      MetadataObject object =
+          MetadataObjects.parse(
+              fullName, MetadataObject.Type.valueOf(type.toUpperCase(Locale.ROOT)));
+      return Utils.doAs(
+          httpRequest,
+          () -> {
+            Optional<Owner> owner = ownershipManager.getOwner(metalake, object);
+            if (owner.isPresent()) {
+              return Utils.ok(new OwnerResponse(DTOConverters.toDTO(owner.get())));
+            } else {
+              return Utils.ok(new MessageResponse("Owner is not set"));
```

Comment:
Whether an empty `OwnerResponse("")` can be return?
Additionally, A RESTful interface can return two different type (Owner|Message) Responses, I think this uncommon, and will increase the burden of the interface caller.

Reply:
I make owner is nullable and delete MessageResponse.

Reply:
OK